radix-engine/src/model/resources/executables/resource_manager.rs

```diff
@@ -72,139 +64,104 @@ impl Executor for ResourceManagerBucketBurnInvocation {
-            RENode::NonFungibleStore(NonFungibleStore::new()),
-        )?;
-        let nf_store_id: NonFungibleStoreId = nf_store_node_id.into();
-
-        let mut resource_manager =
-            ResourceManagerSubstate::new(resource_type, Some(nf_store_id), resource_address)
-                .map_err(|e| match e {
-                    InvokeError::Error(e) => {
-                        RuntimeError::ApplicationError(ApplicationError::ResourceManagerError(e))
-                    }
-                    InvokeError::Downstream(e) => e,
-                })?;
-
-        let bucket = if let Some(mint_params) = mint_params {
-            if let MintParams::NonFungible { entries } = mint_params {
-                for (non_fungible_id, data) in &entries {
-                    if non_fungible_id.id_type() != id_type {
-                        return Err(RuntimeError::ApplicationError(
-                            ApplicationError::ResourceManagerError(
-                                ResourceManagerError::NonFungibleIdTypeDoesNotMatch(
-                                    non_fungible_id.id_type(),
-                                    id_type,
-                                ),
-                            ),
-                        ));
-                    }
+    let nf_store_node_id = api.allocate_node_id(RENodeType::NonFungibleStore)?;
+    api.create_node(
+        nf_store_node_id,
+        RENode::NonFungibleStore(NonFungibleStore::new()),
+    )?;
+    let nf_store_id: NonFungibleStoreId = nf_store_node_id.into();
+
+    let mut resource_manager = ResourceManagerSubstate::new(
+        ResourceType::NonFungible { id_type },
+        Some(nf_store_id),
+        resource_address,
+    );
 
-                    let offset = SubstateOffset::NonFungibleStore(NonFungibleStoreOffset::Entry(
-                        non_fungible_id.clone(),
-                    ));
-                    let non_fungible_handle =
-                        api.lock_substate(nf_store_node_id, offset, LockFlags::MUTABLE)?;
-                    let mut substate_mut = api.get_ref_mut(non_fungible_handle)?;
-                    let non_fungible_mut = substate_mut.non_fungible();
-                    *non_fungible_mut = NonFungibleSubstate(Some(
-                        NonFungible::new(data.0.clone(), data.1.clone()), // FIXME: verify data
-                    ));
-                    api.drop_lock(non_fungible_handle)?;
-                }
-                resource_manager.total_supply = entries.len().into();
-                let ids = entries.into_keys().collect();
-                let container = Resource::new_non_fungible(resource_address, ids, id_type);
-                let node_id = api.allocate_node_id(RENodeType::Bucket)?;
-                api.create_node(node_id, RENode::Bucket(BucketSubstate::new(container)))?;
-                let bucket_id = node_id.into();
-                Some(Bucket(bucket_id))
-            } else {
+    let bucket = {
```

Comment:
Why not create the resource manager node first and then do a `ResourceManagerMintNonFungibleInvocation` to mint the non-fungible tokens? Is this potentionally something to do with auth?

Reply:
Yup, one of the reasons, as well as not wanting to make another invocation to setup the initial state.